src/cinder/block_store.go

```diff
@@ -0,0 +1,259 @@
+package cinder
+
+import (
+	"fmt"
+	"math/rand"
+	"strconv"
+
+	"github.com/Lirt/velero-plugin-swift/src/utils"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	"github.com/gophercloud/gophercloud/openstack/blockstorage/v3/snapshots"
+	"github.com/gophercloud/gophercloud/openstack/blockstorage/v3/volumes"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/vmware-tanzu/velero/pkg/plugin/velero"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// BlockStore is a plugin for containing state for the Cinder Block Storage
+type BlockStore struct {
+	client    *gophercloud.ServiceClient
+	provider  *gophercloud.ProviderClient
+	config    map[string]string
+	volumes   map[string]volumes.Volume
+	snapshots map[string]snapshots.Snapshot
+	log       logrus.FieldLogger
+}
+
+// NewBlockStore instantiates a Cinder Volume Snapshotter.
+func NewBlockStore(log logrus.FieldLogger) *BlockStore {
```

Comment:
Why is initialization in a separate Init funcion? Can't you do Init directly in here?